src/mame/snk/hng64.cpp

```diff
@@ -1113,8 +1145,19 @@ void hng64_state::hng64_sprite_clear_odd_w(offs_t offset, uint32_t data, uint32_
 
 void hng64_state::hng64_vregs_w(offs_t offset, uint32_t data, uint32_t mem_mask)
 {
-	LOGMASKED(LOG_DMA, "hng64_vregs_w %02x, %08x %08x\n", offset * 4, data, mem_mask);
+	uint32_t oldreg = m_videoregs[offset];
+
+	LOGMASKED(LOG_VREGS, "hng64_vregs_w %02x, %08x %08x\n", offset * 4, data, mem_mask);
 	COMBINE_DATA(&m_videoregs[offset]);
+
+	if (oldreg != m_videoregs[offset])
+	{
+		int vpos = m_screen->vpos();
+
+		if (vpos > 0)
+			m_screen->update_partial(vpos - 1);
+	}
+
```

Comment:
Isn’t it usually best to call `update_partial` before updating the registers?  Like:
```c++
	LOGMASKED(LOG_VREGS, "hng64_vregs_w %02x, %08x %08x\n", offset * 4, data, mem_mask);

	uint32_t newval = m_videoregs[offset];
	COMBINE_DATA(&newval);

	if (newval != m_videoregs[offset])
	{
		int vpos = m_screen->vpos();

		if (vpos > 0)
			m_screen->update_partial(vpos - 1);

		m_videoregs[offset] = newval;
	}
```

Reply:
yes, that does make more sense